assets/js/base/utils/errors.js

```diff
@@ -1,22 +1,30 @@
-const formatErrorMessage = ( error, messageProperty = 'frontendMessage' ) => {
-	if ( typeof error === 'object' && error.hasOwnProperty( 'message' ) ) {
-		return {
-			[ messageProperty ]: error.message,
-		};
-	}
-
-	return error;
-};
-
+/**
+ * Given a JS error or a fetch response error, parse and format it so it can be displayed to the user.
+ *
+ * @param {object} error Error object.
+ * @param {function} [error.json] If a json method is specified, it will try parsing the error first.
+ * @param {string} [error.message] If a message is specified, it will be shown to the user.
+ * @param {string} [error.type] The type will be used to determine how the error is shown to the user.
```

Comment:
I'm not sure this description is sufficient.  In implementations and what you've described in your reply to me is this describes _the context in which the error was triggered_?  It's up to code implementing the return from this function to decide what it does with the error (in the one case I saw, the message was always shown, just the prefix before the message output was updated).

Reply:
> I'm not sure this description is sufficient. In implementations and what you've described in your reply to me is this describes the context in which the error was triggered?

Yes, I imagined in the future we might want to have a `user-friendly` type that directly skipped the prefix before the message. But I'm ok rewriting the description now.

Done in 02df7c2bb286f0e89eefb60f67496b067b2d8ee4. I also renamed `frontend` to `general`.